frontends/mit-open/src/page-components/Header/Header.tsx

```diff
@@ -7,48 +7,137 @@ import {
-  height: 56px;
-  padding: 0 8px;
-  z-index: ${({ theme }) => theme.zIndex.drawer + 1};
-  background-color: ${({ theme }) => theme.custom.colors.white};
-  color: ${({ theme }) => theme.custom.colors.black};
-  display: flex;
-  flex-direction: column;
-  box-shadow: 0 2px 10px rgba(120 169 197 / 15%);
-`
+const Bar = styled(AppBar)(({ theme }) => ({
+  height: "80px",
+  padding: "0 8px",
+  zIndex: theme.zIndex.drawer + 1,
+  backgroundColor: theme.custom.colors.white,
+  color: theme.custom.colors.darkGray1,
+  display: "flex",
+  flexDirection: "column",
+  boxShadow: "0 2px 10px rgba(120 169 197 / 15%)",
+  [theme.breakpoints.down("sm")]: {
+    height: "60px",
+    padding: "0",
+  },
+}))
+
+const FlexContainer = styled.div({
+  display: "flex",
+  alignItems: "center",
+})
+
+const DesktopOnly = styled(FlexContainer)(({ theme }) => ({
+  [theme.breakpoints.up("sm")]: {
+    display: "flex",
+  },
+  [theme.breakpoints.down("sm")]: {
+    display: "none",
+  },
+}))
+
+const MobileOnly = styled(FlexContainer)(({ theme }) => ({
+  [theme.breakpoints.down("sm")]: {
+    display: "flex",
+  },
+  [theme.breakpoints.up("sm")]: {
+    display: "none",
+  },
+}))
 
 const StyledToolbar = styled(Toolbar)({
   flex: 1,
 })
 
-const LogoLink = styled(MITLogoLink)({
-  width: 45,
-  height: "auto",
+const LogoLink = styled(MITLogoLink)(({ theme }) => ({
   img: {
-    height: 20,
+    width: 109,
+    height: 40,
+    [theme.breakpoints.down("sm")]: {
+      marginLeft: "16px",
+    },
   },
-})
+}))
 
-const StyledDivider = styled(Divider)({
-  margin: "0.5em 1em",
+const LeftDivider = styled(Divider)({
+  margin: "0 24px",
 })
 
+const RightDivider = styled(Divider)(({ theme }) => ({
+  margin: "0 32px",
+  [theme.breakpoints.down("sm")]: {
+    margin: "0 16px",
+  },
+}))
+
 const Spacer = styled.div`
   flex: 1;
 `
 
+const StyledSearchIcon = styled(RiSearch2Line)(({ theme }) => ({
+  color: theme.custom.colors.darkGray1,
+  margin: "4px 0",
+}))
+
+const SearchButton: FunctionComponent = () => {
+  return (
+    <ActionButtonLink
+      edge="rounded"
+      variant="text"
+      nativeAnchor={true}
+      href={SEARCH}
+    >
+      <StyledSearchIcon />
+    </ActionButtonLink>
+  )
+}
+
+const LoggedOutView: FunctionComponent = () => {
+  return (
+    <FlexContainer>
+      <DesktopOnly>
+        <UserMenu variant="desktop" />
+        <SearchButton />
+      </DesktopOnly>
+      <MobileOnly>
+        <SearchButton />
+        <RightDivider orientation="vertical" flexItem />
+        <UserMenu variant="mobile" />
+      </MobileOnly>
+    </FlexContainer>
+  )
+}
+
+const LoggedInView: FunctionComponent = () => {
+  return (
+    <FlexContainer>
+      <SearchButton />
+      <RightDivider orientation="vertical" flexItem />
+      <UserMenu />
+    </FlexContainer>
+  )
+}
+
+const UserView: FunctionComponent = () => {
+  const { data: user } = useUserMe()
+  return user?.is_authenticated ? <LoggedInView /> : <LoggedOutView />
```

Comment:
Can we check for `!isLoaded` here and return `null`. That will avoid the flash of the logged out view while the user is fetched.

Reply:
Yep, I'll need to take a look at this. I took it out because it was causing issues with testing, but I'll add it back in and try and figure that out.